src/com/projectkorra/projectkorra/GeneralMethods.java

```diff
@@ -2532,4 +2500,51 @@ public static boolean isLightEmitting(final Material material) {
 				return false;
 		}
 	}
+
+	@Deprecated
+	public static void setVelocity(Entity entity, Vector vector) {
+		setVelocity(null,entity,vector);
+	}
+	
+	public static void setVelocity(Ability ability, Entity entity, Vector vector) {
+		
+		final AbilityVelocityAffectEntityEvent event = new AbilityVelocityAffectEntityEvent(ability, entity, vector);
+		Bukkit.getServer().getPluginManager().callEvent(event);
+		if (event.isCancelled()) 
+			return;
+		
+		Vector velocity = event.getNewVector();

```

Comment:
This code is not null safe with respect to `velocity`, it would be a good idea to add a null check and possible a NaN check to `velocity` here.